db/transaction_log_impl.cc

```diff
@@ -234,21 +234,27 @@ void TransactionLogIteratorImpl::UpdateCurrentWriteBatch(const Slice& record) {
   // If the iterator has started, then confirm that we get continuous batches
   if (started_ && !IsBatchExpected(batch.get(), expectedSeq)) {
     // Seek to the batch having expected sequence number
+    bool file_changed = false;
     if (expectedSeq < files_->at(currentFileIndex_)->StartSequence()) {
       // Expected batch must lie in the previous log file
       // Avoid underflow.
       if (currentFileIndex_ != 0) {
         currentFileIndex_--;
+        file_changed = true;
       }
     }
-    startingSequenceNumber_ = expectedSeq;
-    // currentStatus_ will be set to Ok if reseek succeeds
-    // Note: this is still ok in seq_pre_batch_ && two_write_queuesp_ mode
-    // that allows gaps in the WAL since it will still skip over the gap.
-    currentStatus_ = Status::NotFound("Gap in sequence numbers");
-    // In seq_per_batch_ mode, gaps in the seq are possible so the strict mode
-    // should be disabled
-    return SeekToStartSequence(currentFileIndex_, !seq_per_batch_);
+    if (file_changed) {
+      startingSequenceNumber_ = expectedSeq;
+      // currentStatus_ will be set to Ok if reseek succeeds
+      // Note: this is still ok in seq_pre_batch_ && two_write_queuesp_ mode
+      // that allows gaps in the WAL since it will still skip over the gap.
+      currentStatus_ = Status::NotFound("Gap in sequence numbers");
+      // In seq_per_batch_ mode, gaps in the seq are possible so the strict mode
+      // should be disabled
+      return SeekToStartSequence(currentFileIndex_, !seq_per_batch_);
+    } else if (seq_per_batch_) {
+      currentStatus_ = Status::NotFound("Unexpected failures!");
```

Comment:
Why not continue call it "Gap in sequence numbers"?

Reply:
sorry, you mean condition match "!file_changed && seq_per_batch_" that need to call SeekToStartSequence again? 
But in this circumstance, currentFileIndex_ is the same that I think no need to reseek it, just read batch in currentFileIndex_.